pkg/gofr/http/middleware/oauth.go

```diff
@@ -54,6 +54,7 @@ func NewOAuth(config OauthConfigs) PublicKeyProvider {
 	var publicKeys PublicKeys
 
 	go func() {
+		// TODO update the keys when starting
```

Comment:
This is about calling updateKeys without waiting the first ticker.

It will help to provide shorter delay for running the tests

But as the type if refresh interval is now a duration maybe it's not needed.

It's worth a double check

Reply:
As you refactored and used a few milliseconds in the test, this refactoring is no longer needed maybe